File: pset1/cash/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    float cash = 0.0;
    
    //Coins
    int qt = 25; //Quarter
    int dm = 10; //Dime
    int nk = 5;  //Nickel     
    int pn = 1;  //Penny
    
    int coins = 0;

    //Prompt user to input change
    
    do{
        cash = get_float("Change owed\n");
    } while (cash < 0);

    //Convert dollars to cents
    
    int cents = round(cash * 100);
  
    //Change in quarters
    while (cents >= qt){
        coins++;
        (cents -= qt);
    } 

    //Change in dimes
    while (cents >= dm){
        coins++;
        (cents -= dm);
    }
    
    //Change in nickels
    while (cents >= nk){
        coins++;
        (cents -= nk);
    } 
    
    //Change in pennies
    while (cents >= pn){
        coins++;
        (cents -= pn);
    }
    
    printf("Coins returned: %i\n", coins);
}

File: pset6/cash/cash.py
from cs50 import get_float

def main():

    change = get_change()
    cents = change * 100

    ## Coins
    quarter = 25
    dime = 10
    nickel = 5
    penny = 1

    coins = 0

    ##How many quarters?
    while cents >= quarter:

        coins += 1
        cents -= quarter

    ##How many dimes?
    while cents >= dime:

        coins += 1
        cents -= dime

    ##How many nickels?
    while cents >= nickel:

        coins += 1
        cents -= nickel

    ##How many pennies?
    while cents >= penny:

        coins += 1
        cents -= penny

    ##We print the total of coins returned
    print(f"Coins returned: {coins}")

def get_change():

    while True:
        change = get_float("Change owed: ")

    ##User can only input positive values
        if change > 0:
            break
    return change

main()
